packages/notifi-react-card/lib/components/subscription/NotifiSubscriptionCardContainer.tsx

```diff
@@ -23,15 +23,24 @@ export const NotifiSubscriptionCardContainer: React.FC<
   loadingRingColor,
   loadingSpinnerSize,
   onClose,
+  demoByJSONConfig,
 }: React.PropsWithChildren<NotifiSubscriptionCardProps>) => {
   const { isInitialized } = useNotifiSubscribe({ targetGroupName: 'Default' });
-  const { loading } = useNotifiSubscriptionContext();
+  const { loading, setDemoPreview } = useNotifiSubscriptionContext();
+
+  useEffect(() => {
+    demoByJSONConfig ? setDemoPreview(demoByJSONConfig) : setDemoPreview(null);
+  }, [demoByJSONConfig]);
```

Comment:
curious -- any reason we have to set this via useEffect? why can't we just pass it as a prop to the NotifiSubscriptionContext?

Reply:
Migrated demoPreview to newly created 'NotifiDemoPreviewContext' that handles demoPreview related context